Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -27,11 +27,7 @@ class DefaultPaymentViewModel: PaymentViewModel {
     }
   }
 
-  private var cardDetails: CardCreationModel =  CardCreationModel() {
-    didSet {
-      shouldEnablePayButton?(cardDetails.expiryDate != nil && cardDetails.number != nil)
-    }
-  }
+  private var cardDetails: CardCreationModel =  CardCreationModel()
```

Comment:
I had to remove this to avoid duplicate call when setting the card number and scheme. With this approach it would get triggered twice, once for each property.

The change to manually call validation method when deemed necessary allows more control over frequency of those calls, not being tied to the card details being updated necessarily.